### PR TITLE
All changes according to the comments in the excel

### PR DESCRIPTION
**Pull request by Artdianic94**
lombok and email/password in Jenkinsfile haven't been deleted because they may be useful in solving the problem with credentials

### CODE REVIEW

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 1]
```diff
@@ -22,75 +20,66 @@ public class AmazonAddToCartPage extends BasePage {
...
-        iphoneIndex.append(iPhonesXpath);
-        for (int i = 1; i <= allIphones.size(); i++) {
-            iphoneIndex.append("[").append(i).append("]");
-            LOGGER.info("Search for a product that can be added to the Cart");
-            wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(String.valueOf(iphoneIndex))));
-            driver.findElement(By.xpath(String.valueOf(iphoneIndex))).click();
+    @Step("Search for the product that has add button")
+    public void addProductThatHasAddBtn(List<WebElement> allIphones) {
+        for (int i = 0; i < allIphones.size(); i++) {
+            driver.findElements(IPHONE_IN_LIST).get(i).click();
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:37]: why do we need `findElements` each time?

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 2]
```diff
@@ -22,75 +20,66 @@ public class AmazonAddToCartPage extends BasePage {
...
             }
         }
     }
 
+    @Step("Add product to the Cart")
+    public void addToCart() {
+        productAddedToCart = driver.findElement(PRODUCT_TITLE).getText();
+        driver.findElement(ADD_TO_CART_BTN).click();
+        LOGGER.info(String.format("Searched product with a title %s was added in Cart", productAddedToCart));
+
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:55]: empty string

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 3]
```diff
@@ -22,75 +20,66 @@ public class AmazonAddToCartPage extends BasePage {
...
                 LOGGER.warn("The product page doesn't contain a 'Add to Cart button'");
                 driver.navigate().back();
-                iphoneIndex.delete(iphoneIndex.length() - 3, iphoneIndex.length());
             }
         }
     }
 
+    @Step("Add product to the Cart")
+    public void addToCart() {
+        productAddedToCart = driver.findElement(PRODUCT_TITLE).getText();
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:52]: we find element only for logging product name - it is redundant

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 4]
```diff
@@ -22,75 +20,66 @@ public class AmazonAddToCartPage extends BasePage {
...
+        if (driver.findElement(TICK_ICON).isDisplayed()) {
+            isTickIconGreen();
+        }
+    }
+
+    @Step("Get a simple message after adding the product to the cart")
+    public void getAnotherTypeOfMessage() {
+        alertText = driver.findElement(ANOTHER_ALERT_MESSAGE).getText();
+        if (driver.findElement(ANOTHER_TICK_ICON).isDisplayed()) {
+            isTickIconGreen();
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:70]: what is the purpose of this method? `isTickIconGreen()` is not used anywhere

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 5]
```diff
@@ -105,11 +94,18 @@ public String numberOnCart() {
 
     @Step("Getting a colour of the tick when product was added to the Cart ")
     public boolean isTickIconGreen() {
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:96]: method name is `isTickIconGreen` but it is checking text "Added"

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 6]
```diff
@@ -105,11 +94,18 @@ public String numberOnCart() {
...
+        boolean isTickIconGreen;
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:97]: the following 8 lines may be replaced with just `return alertText.contains("Added");`

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 7]
```diff
@@ -22,75 +20,66 @@ public class AmazonAddToCartPage extends BasePage {
...
-    List<WebElement> deleteAllProductsInCart;
-    StringBuilder deleteIndex = new StringBuilder();
-    boolean isTickIconGreen = false;
-    String iPhonesXpath = "(//div[@class='a-section']//span[contains(text(), 'iPhone')])";
     private final By ADD_TO_CART_BTN = By.id("add-to-cart-button");
-    StringBuilder iphoneIndex = new StringBuilder();
-    List<WebElement> allIphones;
-    static String productAddedToCart;
-    AmazonSearchPage amazonSearchPage;
+    private final By IPHONE_IN_LIST = By.xpath("(//div[@class='a-section']//span[contains(text(), 'iPhone')])");
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:24]: there is the same locator on AmazonSearchPage

File: src/test/java/tests/AmazonProductInCartTest.java [hunk 8]
```diff
@@ -18,22 +15,14 @@ public void loginAndOpenProduct() {
...
-        amazonAddToCartPage.checkForAddingToCart();
+        amazonAddToCartPage.addProductThatHasAddBtn(amazonSearchPage.getListOfSearchProduct(productName));
+        amazonAddToCartPage.getTextFromMessage();
     }
 
     @Test
     @Description(value = "The test checks that the Cart contains the added Phone")
     public void checkProductsInCartTest() {
-        boolean actualProduct = amazonAddToCartPage.whatInCart();
+        boolean actualProduct = amazonAddToCartPage.doesCartContainSelectedProduct();
```
Comment (rmarchuk) [inline src/test/java/tests/AmazonProductInCartTest.java:25]: name of variable doesn't represent its content. we can just don't use variable here at all

**Review by rmarchuk — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 9]
```diff
@@ -34,12 +34,12 @@ dependencies {
             "org.junit.jupiter:junit-jupiter:$junitVersion")
     testRuntimeOnly('org.slf4j:slf4j-simple:1.7.33',
             "org.junit.jupiter:junit-jupiter-engine:$junitVersion")
+    compileOnly 'org.projectlombok:lombok:1.18.26'
+    annotationProcessor 'org.projectlombok:lombok:1.18.26'
+
+    testCompileOnly 'org.projectlombok:lombok:1.18.26'
+    testAnnotationProcessor 'org.projectlombok:lombok:1.18.26'
 }
-//def amazonTest = tasks.register("amazonTest", Test) {
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply build.gradle:38]: commented lines were deleted

File: src/test/java/pages/BasePage.java [hunk 10]
```diff
@@ -8,4 +9,9 @@ public class BasePage {
     public BasePage(WebDriver driver) {
         this.driver = driver;
     }
+
+    @Step("Open main page {amazonUrl}")
+    public void openMainPage(String amazonUrl) {
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/pages/BasePage.java]: method name was changed

File: src/test/java/utilities/AfterEachExtension.java [hunk 11]
```diff
@@ -18,7 +17,7 @@ public void setDriver(WebDriver driver) {
     }
 
     @Override
-    public void afterEach(ExtensionContext context) {
+    public void afterEach(ExtensionContext context) throws IOException {
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/utilities/AfterEachExtension.java]: exception was deleted

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 1 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: fixed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 2 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: empty string was deleted

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 4 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: this method was changed completely

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 5 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: fixed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 6 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: fixed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 7 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: fixed

File: src/test/java/tests/AmazonProductInCartTest.java (hunk 8 above)
Comment (Artdianic94) [reply src/test/java/tests/AmazonProductInCartTest.java]: fixed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 3 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: fixed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 4 above)
Comment (rmarchuk) [reply src/test/java/pages/AmazonAddToCartPage.java]: it is better to rename the method to `setAnotherTypeOfMessage` as the return type is void

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 7 above)
Comment (rmarchuk) [reply src/test/java/pages/AmazonAddToCartPage.java]: I still can see duplicate

**Comment by rmarchuk** [comment]
please add messages to all assertions in your tests

**Review by rmarchuk — CHANGES_REQUESTED** [review]
please add some more changes according to comments

File: src/test/java/pages/AmazonAddToCartPage.java [hunk 12]
```diff
@@ -103,26 +90,35 @@ public String numberOnCart() {
         return numberOnCart;
     }
 
-    @Step("Getting a colour of the tick when product was added to the Cart ")
-    public boolean isTickIconGreen() {
-        return isTickIconGreen;
+    @Step("Getting an area of the image with a tick")
+    public String getImageAreaWithTick() {
+        try {
+            return driver.findElement(TICK_ICON_IN_ALERT).getCssValue("background-position");
```
Comment (rmarchuk) [inline src/test/java/pages/AmazonAddToCartPage.java:96]: it is better to check `success` class of element

**Comment by rmarchuk** [comment]
rename AddToCartPage -> ProductPage

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 4 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: methods names were changed

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 7 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: duplicates were excluded

File: src/test/java/pages/AmazonAddToCartPage.java (hunk 12 above)
Comment (Artdianic94) [reply src/test/java/pages/AmazonAddToCartPage.java]: Method with Tick was changed

**Review by rmarchuk — APPROVED** [review]

**Pull request merged**